common/protob/messages-management.proto

```diff
@@ -20,7 +20,7 @@ enum BackupType {
  * @next Features
  */
 message Initialize {
-    optional bytes session_id = 1;           // assumed device session id; Trezor clears caches if it is different or empty
+    optional bytes session_id = 3;           // assumed device session id; Trezor clears caches if it is different or empty
```

Comment:
is this necessary? the `state` field that was 1 originally is functionally identical imo

Reply:
I agree, but this is cleaner and more explicit. If wallet pushed old field `1` Trezor will know it does not even have to check the field, because it comes from the old wallet. If there is a strong argument to keep `1` you'll be able to persuade me, though :-)

Reply:
Here's an argument: if the id stays the same, I can write code that works with fw < 2.3.0 and fw >= 2.3.0 at the same time, without having a "if version < (2,3,0)" check

pre 2.3.0:
```
-> Initialize(session_id=ABCD)
<- Features(session_id=None)
-> GetAddress
<- PassphraseRequest(on_device=True)
-> PassphraseAck(on_device=True, passphrase=None, state=ABCD)
<- PassphraseStateRequest(state=EFGH)
-> PassphraseStateAck
<- Address
...
-> Initialize(session_id=EFGH)
```

2.3.0 and up:
```
-> Initialize(session_id=ABCD)
<- Features(session_id=EFGH)
-> GetAddress
<- PassphraseRequest
-> PassphraseAck(on_device=True, passphrase=None, state=EFGH)
<- Address
...
-> Initialize(session_id=EFGH)
```

I mean, to make it work I need to muck about with protobuf definitions locally anyway, so I can as well add back the `state = 1` and send both. But then the message will be twice as big.

If 2.3.0 is not a mandatory update, having such code in trezorlib would be useful, esp. for Electrum where it's difficult to tell people to "match your trezorlib version to your firmware version"

Reply:
What is the situation for T1? You'd need to have two different code paths for < 1.9.0 and >= 1.9.0, right?

Reply:
T1 doesn't use any of these fields so I don't need to do anything special AFAICT. I decide whether to send `on_device=True` based on capabilities, which comes out right, and other than that the T1 flow is identical across versions.

Reply:
speaking of, this might even be an argument for leaving the old definitions for fields and `PassphraseState` messages in the proto files (so that users don't need to hack the protobuf locally)